### ISSUE TITLE
Implement https connection support on httplistener on Unix

### ISSUE THREAD

**Issue by Priya91** [dotnet/corefx#14691]
This requires a mechanism to get the ssl certificate info to use on the port of httplistener connection. Expose an API or maintain a keystore on Unix.

**Comment by CIPop** [comment]
/cc @davidsh @DavidGoll 
@Priya91 on Mono, HttpListener is searching for certs in a hardcoded path: https://github.com/mono/mono/blob/3f779475e3fc982e312212d5dbbd86515eddfc0c/mcs/class/System/System.Net/HttpListener.Mono.cs#L64
(also see this [Stack Overflow discussion](https://stackoverflow.com/questions/23243224/how-to-use-ssl-with-httplistener-with-an-mkbundled-mono-app)).

I couldn't find any documentation on how this could be achieved on Xamarin.

**Comment by Priya91** [comment]
@CIPop Yes, that's the second bullet in the description. We could re-use the dotnet crypto cert-store for this purpose, similar to mono, instead the hardcoded path here will be some custom dotnet path.

**Comment by DavidGoll** [comment]
@davidsh Document this as a compat issue for UWP6.0. This will require significant design for a cross-plat implementation.

**Comment by davidsh** [comment]
Added documentation to https://github.com/dotnet/corefx/wiki/ApiCompat

**Comment by m5x** [comment]
It would be great if it was possible to supply the certificate as a stream. It would remove additional steps that are required on each OS to install the certficate and at the same time it would allow developers to control themselves how and where is the certificate stored.
